rustls/src/msgs/deframer.rs

```diff
@@ -732,4 +762,126 @@ mod tests {
         );
         assert!(input_bytes(&mut d, &message).is_err());
     }
+
+    #[test]
+    fn test_plaintext_alert_pre_encrypted() {
```

Comment:
I flailed around for a while trying to figure out the best way to test this change and ultimately landed on this unit test approach with carefully mocked `MessageDecrypter`s. 

Initially I wanted to write an integration test in `tests/api.rs` - this ends up being really challenging because Rustls (sensibly) doesn't expose low level guts for writing arbitrary TLS alerts. We can send close notify alerts through the public API, but these are encrypted based on the connection's encryption state (the right behaviour!). 

To be able to tickle the code path in question we need a way to send a plaintext alert when the remote peer has its record layer set up in a state that expects to decrypt messages and I simply couldn't think of a nice way to do that from an integration test. Maybe someone else will have a clever idea?

Reply:
Try something like:
```
    let (mut client, _) = make_pair(KeyType::Rsa);
    client
        .read_tls(&mut io::Cursor::new(&[
            0x15, 0x03, 0x03, 0x00, 0x02, 0x02, 0x6f,
        ]))
        .unwrap();
    assert_eq!(
        client.process_new_packets(),
        Err(Error::AlertReceived(
            rustls::AlertDescription::CertificateUnobtainable
        )),
    );
```